utils/ixdtf/src/parsers/mod.rs

```diff
@@ -340,23 +341,31 @@ impl<'a> Cursor<'a> {
 
     /// Peek the value at next position (current + 1).
     fn peek(&self) -> Option<char> {
-        self.peek_n_char(1)
+        self.peek_n(1)
     }
 
     /// Returns current position in source as `char`.
     fn current(&self) -> Option<char> {
-        self.peek_n_char(0)
+        self.peek_n(0)
     }
 
-    /// Peek the value at n len from current.
-    fn peek_n(&self, n: usize) -> Option<&'a str> {
-        let index = self.pos + n;
-        self.source.get(index..index + 1)
+    /// Peeks the value at `n` as a `char`.
+    fn peek_n(&self, n: usize) -> Option<char> {
+        let (_, char) = self.peek_with_info(n)?;
+        Some(char)
     }
 
-    /// Peeks the value at `n` as a `char`.
-    fn peek_n_char(&self, n: usize) -> Option<char> {
-        self.peek_n(n).map(str::chars).and_then(|mut c| c.next())
+    /// Peeks the value at `n` and returns the char with its byte length.
+    fn peek_with_info(&self, n: usize) -> Option<(usize, char)> {
+        let mut chars =
+            Utf8CharIndices::new(self.source.get(self.pos..self.source.len()).unwrap_or(&[]));
```

Comment:
I mentioned this on a previous comment, but there's no erroring version of `Utf8CharIndices` currently. Is there any preference towards erroring on invalid utf8?

Reply:
I think the idea is that ill-formed UTF-8, even if it resolves to the replacement character, won't be valid IXDTF syntax since the replacement character is not valid in any part of the grammar.

Reply:
Yep! That's the current idea. The replacement character will fail the grammar test currently, but that's a little different than flagging the character as invalid.

For instance, say there is an invalid character in a year. The error thrown currently would be `ParserError::DateYear` vs. potentially throwing a `ParserError::Utf8CharError`. Not sure if there'd be a preference there.

Reply:
I see, hmm.